### PR TITLE
[SW-2525] Expose Interactions on Target Encoder

### PR DESCRIPTION
**Pull request by mn-mikke**
The pr depends on https://github.com/h2oai/h2o-3/pull/5231

### CODE REVIEW

**Review by sebhrusen — COMMENTED** [review]
looks good: minor comments.

File: py/src/ai/h2o/sparkling/ml/features/H2OTargetEncoder.py [hunk 1]
```diff
@@ -37,17 +37,33 @@ def __init__(self,
                  blendedAvgInflectionPoint=10.0,
                  blendedAvgSmoothing=20.0,
                  noise=0.01,
-                 noiseSeed=-1):
+                 noiseSeed=-1,
+                 columnsToEncode=None):
```
[the remark this answers is not in the document]
Comment (mn-mikke) [reply py/src/ai/h2o/sparkling/ml/features/H2OTargetEncoder.py:41]: Good catch, thanks! This is a leftover from the initial version of this PR

File: scoring/src/main/scala/ai/h2o/sparkling/ml/params/H2OTargetEncoderMOJOParams.scala [hunk 2]
```diff
@@ -86,12 +89,12 @@ trait H2OTargetEncoderMOJOParams extends Params {
...
   def getLabelCol(): String = $(labelCol)
 
-  def getInputCols(): Array[String] = $(inputCols)
+  def getInputCols(): Array[Array[String]] = $(inputCols)
 
   def getOutputCols(): Array[String] = {
     val columns = $(outputCols)
     if (columns.isEmpty) {
-      getInputCols().map(_ + "_te")
+      getInputCols().map(_.mkString("~") + "_te")
```
[the remark this answers is not in the document]
Comment (mn-mikke) [reply scoring/src/main/scala/ai/h2o/sparkling/ml/params/H2OTargetEncoderMOJOParams.scala:97]: good idea, thanks!

**Comment by mn-mikke** [comment]
@sebhrusen and @michalkurka do you have any further comments to this PR?

**Review by sebhrusen — APPROVED** [review]
Glad to see this in SW, thanks @mn-mikke !

**Review by michalkurka — APPROVED** [review]
Thank you @mn-mikke 

We should popularize this feature as well as TE itself, it would be useful even for people who do not end up using our algos.

**Pull request merged**